detox/e2e/test/channel_info/channel_notification_default.e2e.js

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```

Comment:
Should this be located at `detox/e2e/test/notifications` folder?

Reply:
Why notifications folder?

Reply:
It's to match folder structure (or functional area) of a given TM4J test case.
![Screen Shot 2020-11-05 at 5 52 11 PM](https://user-images.githubusercontent.com/5334504/98225509-bdcc4f80-1f8f-11eb-9290-e77572d1441b.png)

Reply:
I fear this is not really scalable specially cause test cases may not even be in TM4J at the time they are written, the same can be said about the ID. In fact it has already happened to me and outside contributors so now adding a folder structure to go along side it seems too far fetched in my opinion

Reply:
I may agree on folder structure but
a. I'm incline to have some basis in doing so rather than spreading test cases all over the place. We started like that in Cypress and the folder/file structure ended up messy. Naming folder is not easy, subjective and no right/wrong. I find the TM4J folder structure to be a better guide in this regard. But it might only be me.
b. Consider running tests in group. If I'd like to run all notifications tests and are found in a particular folder, then one option currently is to setup Jest's `testRegex` or `testMatch`. (Though, we'll try to find other solution as we did in Cypress).

I'd like to imagine that the only source of truth for E2E is the TM4J. IMO, below are some reasons on why there should have TM4J ID before accepting E2E tests:
- To ensure a test case is written/documented and not forgotten.
- To qualify a test case. Is it valid, not duplicate from existing test case or has existing test case to update?
- To be able to save/record each (expensive) test execution once we have Detox/TM4J integration in place.
- To ensure that high overhead test maintenance on flaky tests are well spent on something that we can easily measure or link to issue.

Reply:
I agree with all your points 100% I just worry that this will be cumbersome for some people, we should:
* Document the folder structure as we did the TM4J_ID
* Have an easy an effective way for contributors (staff or community) to quickly identify them to not hold back feature development and adding the test cases.

other than that I'm ok with the requirements.

Reply:
Thanks and yes, will clarify in the docs (will submit together with other items in my list). For HW, test folder is included in the description.

Reply:
My bad. I had it at the back of my mind to move to notifications but completely forgot.
Moved the tests to `notifications` folder.
Also renamed `notification.e2e.js` to `in_app_notification.e2e.js` and moved to `notifications` folder.